satpy/readers/ahi_hsd.py

```diff
@@ -455,9 +455,12 @@ def _modify_observation_time_for_nominal(self, observation_time):
         else:
             observation_frequency_seconds = {'JP': 150, 'R3': 150, 'R4': 30, 'R5': 30}[self.observation_area[:2]]
             dt = observation_frequency_seconds * (int(self.observation_area[2:]) - 1)
-        return observation_time.replace(
-            hour=int(timeline[:2]), minute=int(timeline[2:4]) + dt//60,
-            second=dt % 60, microsecond=0)
+        if int(timeline[:2]) < 24:
```

Comment:
Could you please move the comparison (hour < 24) to a self-describing method, so that the if statement becomes (for example) `if _is_valid_timeline(timeline)`?

Reply:
Ok, test added and I've also added the function you suggest.